robot-mode.el

```diff
@@ -238,17 +252,20 @@ Used as `indent-line-function' of the mode."
 		 robot-mode-basic-offset)))
 
     ;; Toggle indentation if the line is already indented
-    (when (and (> indent 0)
-	       (= indent current-indent))
-      (setq indent 0))
+    (when robot-mode-indent-toggle 
+      (when (and (> indent 0)
+		 (= indent current-indent))
+	(setq indent 0)))
 
-    ;; Always move back to indentation
-    (back-to-indentation)
+    (when (not robot-mode-retain-point-on-indent)
+      (back-to-indentation))
 
-    ;; Do the actual indenting if indentation changed
-    (when (not (= indent current-indent))
-      (delete-region (line-beginning-position)  (point))
-      (indent-to indent))))
+    ;; Save point if point retention is enabled and point is after indentation
+    (if (and (> point-in-line indent)
```

Comment:
> This is not correct, should be compared with the current indentation and not the calculated indentation. This cause the cursor at column < 4 to go back at indentation when toggling from indent=0 to indent=4 (given an indent offset to 4)

Isn't that how it's supposed to work?
IIUC e.g. in `emacs-lisp-mode` if the point is between `0` and indent offset, and you press `TAB`, then the point moves back to indentation. Or am I misunderstanding something ?

Added usage of `(current-column)` and `(current-indentation)`. Thanks!

I looked at the Emacs source for `current-column` when the documentation said that it was introduced/changed around version 29.1, but apparently that was the variable and not the function. The function has been present for ages.

Reply:
This is absolutely correct, the cursor should follow the indentation when it is located within the current indentation when you press tab.

But here, it checks if it is within the indentation after setting the indentation. Here are some examples to illustrate the "problem":

1. In this case, it should follow the indentation (which is the case)

```robotframework
Test Case 1
| Keyword  
```
And become:
```robotframework
Test Case 1
    |Keyword  
```

2. In this second case, it shouldn't follow indentation and should save the excursion
```robotframework
Test Case 1
Key|word  
```
And should become:
```robotframework
Test Case 1
    Key|word  
```

Which is not the case currently, because it is compared with the `indent` (meaning, the calculated `indent`) and not the `current-indent`. 
From the last example:
  - `current-column = 3`
  - `current-indentation = 0` 
  - `indent (calculated-indent) = 4`
  - `current-column < indent` -> Follow indentation.

Reply:
Thanks for the explanation!
I was able to reproduce the situation, and apply the suggested fix :+1: